lib/gpu/lib/src/buffer.dart

```diff
@@ -0,0 +1,73 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ffi';
+import 'dart:nativewrappers';
+import 'dart:typed_data';
+
+import 'context.dart';
+
+/// A buffer data range.
+class BufferView {
+  /// The buffer of this view.
+  final Buffer buffer;
+  /// The start of the view, in bytes starting from the beginning of the
+  /// [buffer].
+  final int offsetInBytes;
+  /// The end of the view
+  final int lengthInBytes;
+
+  /// Create a new view into a buffer on the GPU.
+  const BufferView(
+      {required this.buffer,
```

Comment:
nit: make the buffer argument positional, leave the others are required named parameters.

Reply:
Done. Your suggestion is nicer IMO, but do we happen to have any guidelines for this kind of thing I should seek to align myself with?

Reply:
I'm not sure that there is a hard and fast rule for this, but I would think about named parameters in cases where the literal value or variable name itself does not give enough clarirty on what the argument is. The canonical example would be something like:

`void foo(true, false)`

What the heck are true / false doing here? So similar to how we might add a comment in the c++ code like 

`void foo(/*arg_name=*/true, /*arg_name=*/false)`, then we can use named arguments to make the Dart code clearer:

`void foo(argName: true, argName2, false)`.

So using the buffer constructor argument above, we've got something like:

`BufferView(buffer, 0, 12)`. The buffer itself is obvious, but the numbers benefit from clarification.